syntax = "proto3";
package pt.ulisboa.tecnico.tuplespaces.nameserver.contract;

message LookupRequest {
    string ServiceName = 1;
    string Qualifier = 2;
}

message LookupResponse {
    message ServiceEntry {
        string Qualifier = 1;
        string ServiceAddress = 2;
    }
    repeated ServiceEntry ServiceEntries = 1;
}

message RegisterRequest {
    string ServiceName = 1;
    string Qualifier = 2;
    string Address = 3;
}

message RegisterResponse {
    // empty
}

message DeleteRequest {
    string ServiceName = 1;
    string Address = 2;
}

message DeleteResponse {
    // empty
}

service NameServer {
    rpc Register(RegisterRequest) returns (RegisterResponse);
    rpc Lookup(LookupRequest) returns (LookupResponse);
    rpc Delete(DeleteRequest) returns (DeleteResponse);
}